snowplow_tracker/subject.py

```diff
@@ -173,3 +173,15 @@ def set_network_user_id(self, nuid: str) -> "Subject":
         """
         self.standard_nv_pairs["tnuid"] = nuid
         return self
+
+    def combine_subject(self, subject: Optional["Subject"]) -> "Subject":
+        """
+        Merges another instance of Subject, with self taking priority
+        :param  subject     Subject to update
+        :type   subject     subject
+        :rtype              subject
+
+        """
+        if subject is not None:
+            subject.standard_nv_pairs.update(self.standard_nv_pairs)
```

Comment:
When we update the subject dictionary in place that means that the subject will keep the values also after the event is tracked. In most cases this is okay, but I can imagine that some people might reuse the event subject instance and might be confused when it changes after tracking events.

I think it would be better to return a new dictionary from this function that combines the two dictionaries without modifying the original subjects, like this:

```suggestion
            return {**subject.standard_nv_pairs, **self.standard_nv_pairs}
```